neural_network_lyapunov/examples/tinydiffsim/learn_ca_dyn.py

```diff
@@ -0,0 +1,144 @@
+import neural_network_lyapunov.worlds as worlds
+import neural_network_lyapunov.utils as utils
+import neural_network_lyapunov.control_affine_system as mut
+
+import pytinydiffsim as pd
```

Comment:
BTW, why we need tiny diff sim? I don't immediately see why we need to take the gradient in the dynamics equations.

Reply:
For no reason in this particular example, but they support autodiffing through their contact resolution. I think this would make learning a continuous time model of systems that have contact possible (down the line). Do you have other suggestions? Mujoco requires a license, Bullet is basically the same physics as Tiny but minus autodiff, Brax seems pretty limited, Isaac is overkill and closed source, there's Drake? Otherwise Tiny also has "no dependency" (sort of).

Reply:
I am not exactly sure why we need the dynamics gradient to learn contact dynamics. Could you explain more on this?

Anyway, the PR looks good and I will merge it. My reservation is that for the acrobot dynamics used in this example, we don't need a differentiable simulator; we can write the dynamics analytically. And for the other systems that I think we will use very soon (like quadrotor, RC car), I don't think we have to use a simulator either.